eosio_build.sh

```diff
@@ -30,60 +30,113 @@
 # https://github.com/EOSIO/eos/blob/master/LICENSE.txt
 ##########################################################################
 
-	VERSION=1.2
-	ULIMIT=$( ulimit -u )
-	WORK_DIR=$PWD
-	BUILD_DIR=${WORK_DIR}/build
-	TEMP_DIR=/tmp
+	CWD="$( cd "$( dirname "${BASH_SOURCE[0]}" )" && pwd )"
+	if [ "${CWD}" != "${PWD}" ]; then
+		printf "\\n\\tPlease cd into directory %s to run this script.\\n \\tExiting now.\\n\\n" "${CWD}"
+		exit 1
+	fi
+	if [ -f "${PWD}/CMakeCache.txt" ]; then
+		printf "\\n\\tPlease remove file %s/CMakeCache.txt before building EOS.IO.\\n \\tExiting now.\\n\\n" "${PWD}"
+		exit 1
+	fi
+
+   	function usage()
+   	{ 
+		printf "\\tUsage: %s [Build Option -o <Debug|Release|RelWithDebInfo|MinSizeRel>] [CodeCoverage -c ] [Doxygen -d]\\n\\n" "$0" 1>&2
+		exit 1
+	}
+
 	ARCH=$( uname )
+	BUILD_DIR="${PWD}/build"
+	CMAKE_BUILD_TYPE=Release
 	DISK_MIN=20
+	DOXYGEN=false
+	ENABLE_COVERAGE_TESTING=false
+	TEMP_DIR="/tmp"
 	TIME_BEGIN=$( date -u +%s )
-	DOXYGEN=false #set to true to build docs
+	VERSION=1.2
 
 	txtbld=$(tput bold)
 	bldred=${txtbld}$(tput setaf 1)
 	txtrst=$(tput sgr0)
 
+	if [ $# -ne 0 ]; then
+		while getopts ":cdo:" opt; do
+			case "${opt}" in
+				o )
+					options=( "Debug" "Release" "RelWithDebInfo" "MinSizeRel" )
```

Comment:
Other shell files got spaces instead of tabs, this has tabs instead of spaces, it looks really weird. Are you planning to standardize the coding standards?

Reply:
@pacificcode zap the tabs?